Multiple improvements to actions documentation (#335)
packit/packit.dev: content/docs/actions.md

You can probably find yourself in a situation where some part of the packit workflow needs to be
tweaked for your package.

Packit supports actions, a way to change the default implementation for a
command of your choice.  Packit is able to execute multiple commands. Each
action accepts a list of commands. By default, the commands are executed
directly and not in a shell - if you need a shell, just wrap your command like
this: `bash -c "my fancy $command | grep success"`. It's important to quote the
content of the `-c` option so shell interprets it correctly as a single option

> input.

Author: good suggestion! I added more examples and improved your wording a bit

You can probably find yourself in a situation where some part of the packit workflow needs to be
tweaked for your package.

> Packit supports actions, a way to change the default implementation for a
> command of your choice. Packit is able to execute multiple commands for a
> single action. Each action accepts a list of commands. By default, the commands

Reviewer: To avoid overloading the term *command*.
```suggestion
Packit supports actions, which can be used to change the default implementation of some of the steps in the workflow. Packit is able to execute multiple commands for a
single action. Each action accepts a list of commands. By default, the commands
```

Author: neat! I wasn't happy about the initial wording, yours fits well

name. The best practice is to do it from the last step and print it: `bash -c
'echo path/to/archive-$VERSION.tar.gz'`.

If you can, please place the generated archive in the same directory as your

> spec file.

Reviewer 2: btw, it seems it is not needed with packit v0.35+ (tested with sources in a subdir).

Author: for single-source projects, it's true it's not needed but preferred since it may save you some trouble when generating it

hence the wording "do it if you can" :)

Reviewer 2: even with multiple sources :) but you are right, the "if you can" is good.

You can instead use the `sed` program to set the new Sources correctly, e.g.
```
actions:
  fix-spec-file:
  # define one of the Source variables correctly
  - sed -i my_specfile_path -e "s/https.*only-vendor.tar.xz/my_correct_tarball_path/"
  # fill in %release as if packit would have done it
  - bash -c "sed -i my_specfile_path -r \"s/Release:(\s*)\S+/Release:\1${PACKIT_RPMSPEC_RELEASE}%{?dist}/\""
```

### Environment variables set by packit

Additionally, packit sets a few env vars for specific actions.

**fix-spec-file**

`PACKIT_PROJECT_VERSION` — current version of the project (coming from `git describe`)  
`PACKIT_PROJECT_COMMIT` — commit hash of the top commit  
`PACKIT_PROJECT_ARCHIVE` — expected name of the archive  
`PACKIT_RPMSPEC_RELEASE` — value for spec file's `%release` field which packit would set  

**create-archive**

`PACKIT_PROJECT_VERSION` — current version of the project (coming from `git describe`)
`PACKIT_PROJECT_NAME_VERSION` — current name and version of the project (coming from `git describe`)

If you want to see the content of those variables, you can print using `echo`
in the specific action:
```
actions:
  fix-spec-file:

>   - bash -c "PACKIT_PROJECT_VERSION=${PACKIT_PROJECT_VERSION}"

Reviewer: ```suggestion
  - bash -c "echo PACKIT_PROJECT_VERSION=${PACKIT_PROJECT_VERSION}"
```

Author: my copy-pasting skills are atrocious this week :see_no_evil:
